src/core/config.py

```diff
@@ -69,13 +69,37 @@ class OperatorConfig:
     label: str
     parameters: List[OperatorParameters]
 
+@dataclass
+class PostgreParameters:
+    table_names: List[dict]
+
+@dataclass
+class PostgreSQLConfig:
+    label: str
+    type: str
+    parameters: PostgreParameters
+
+@dataclass
+class AmazonQueueParameters:
+    host_name: str
+    queues: List[dict]
+
+
+@dataclass
+class AmazonQueueConfig:
+    label: str
+    type: str
+    parameters: AmazonQueueParameters
+
 
 @dataclass
 class Config:
     store: Optional[StoreConfig]
     queue: Optional[QueueConfig]
     server: Optional[ServerConfig]
     operators: Optional[OperatorConfig]
+    postgresql: Optional[PostgreSQLConfig]
```

Comment:
wait, postgresql and amazon_queue are types of store and queue. their config needs to be handled within store and queue, not as an additional key.

Reply:
so, a lot of changes for that will be required like
lets say if I wish to add `postgres` inside `store`, this means, I will have to create two parts inside `store`, called `es` and `postgresql`. So we will have to change things in `core feluda`, the `es_store.py` file, as the config is called at many places to create the indexes. 

Wherever the config was called to create the indices, say `feluda.config.store.parameters`, will have to be changed to `feluda.config.store.es.parameters`. 

Something similar will also look for queue.

I am willing to make theses changes, but quite some time will be needed, as thorough testing is needed at each step.

Reply:
no not really. See look at how its done here - https://github.com/tattle-made/feluda/blob/main/src/core/store/__init__.py

the way there is this dict
```python
stores = {"es_vec": es_vec.ES}
```
you would add an entry for postgres. Something like
```python
stores = {
    "es_vec": es_vec.ES
    "postgres": Postgres
}
``` 

the key is to find the commonalities between the store modules. Most likely all stores will have a hostname, username, and password. if there are any store specific field (lets say ssl config for mq, then they are passed as is to the constructor of that entity and it will handle it. the other entities will ignore it). these specific fields will be documented.


>... feluda.config.store.parameters, will have to be changed to feluda.config.store.es.parameters.

This is exactly the type of code we should not be writing. If we need to create indices from the worker, we should only be calling feluda.store.create_indices. either we will mandate that this function always exists for a store module or we ensure worker only calls this function if its appropriate.

We can do a pairing session to resolve this if that helps.